internal/collector/eth_get_balance.go

```diff
@@ -0,0 +1,55 @@
+package collector
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type EthGetBalance struct {
+	rpc     *rpc.Client
+	address string
+	desc    *prometheus.Desc
+}
+
+func NewEthGetBalance(rpc *rpc.Client, address string) *EthGetBalance {
+	return &EthGetBalance{
+		rpc:     rpc,
+		address: address,
+		desc: prometheus.NewDesc(
+			"eth_get_balance",
+			"get balance",
+			nil,
+			nil,
+		),
+	}
+}
+
+func (collector *EthGetBalance) Describe(ch chan<- *prometheus.Desc) {
+	ch <- collector.desc
+}
+
+func (collector *EthGetBalance) Collect(ch chan<- prometheus.Metric) {
+	var result hexutil.Big
+	if err := collector.rpc.Call(&result, "eth_getBalance", common.HexToAddress(collector.address), "latest"); err != nil {
+		ch <- prometheus.NewInvalidMetric(collector.desc, err)
+		return
+	}
+
+	i := (*big.Int)(&result)
+	value := toEther(i)
+	valueFloat64, _ := value.Float64()
```

Comment:
Maybe it's a better idea to have the result in WEI. I don't think balance will fluctuate by 10 ETH from one moment to the next.

Reply:
Actually I was thinking on the end value and how is going to be displayed. But we can save WEI and do the conversion in the frontend/consumer.